### ISSUE TITLE
`Clause::$hash` should be `string`

### ISSUE THREAD

**Issue by weirdan** [vimeo/psalm#7706]
I suspec the following should be actually annotated as `string` (unconfirmed yet).

https://github.com/vimeo/psalm/blob/ad91df5ee19923053149120d1764be71766a9d48/src/Psalm/Internal/Clause.php#L80-L81

**Comment by orklah** [comment]
Yeah, it seems to be either the result of a concatenation or an md5
